components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -1439,6 +1490,43 @@ func (p *pg) DeleteProject(ctx context.Context, id string) error {
 	return nil
 }
 
+// ErrIfMissingProjects returns projectassignment.ProjectsMissingErr if there projects missing,
+// otherwise it returns nil.
+func (p *pg) ErrIfMissingProjects(ctx context.Context, projectIDs []string) error {
+	return p.errIfMissingProjectsWithQuerier(ctx, p.db, projectIDs)
+}
+
+func (p *pg) errIfMissingProjectsWithQuerier(ctx context.Context, q Querier, projectIDs []string) error {
+	// Return any input ID that does not exist in the projects table.
+	rows, err := p.db.QueryContext(ctx,
+		`SELECT id FROM unnest($1::text[]) AS input(id)
```

Comment:
What does `input(id)` do?

Reply:
It's the same as other `AS x` statements except it assigns the results of the unnest to the field ID. So you can say `input.id` in the rest of the statement.